ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ConnectExecutor.java

```diff
@@ -62,4 +68,30 @@ private ConnectExecutor() { }
     return response.error()
         .map(err -> new ErrorEntity(statement.getStatementText(), err));
   }
+
+  private static void throwIfConnectorIsDuplicate(final CreateConnector createConnector,
+      final ConnectClient client) {
+    if (createConnector.isNotExists()) {
+      checkConnectorExists(client, createConnector)
+          .ifPresent(connector -> {
+            throw new KsqlException(String.format(
+                "Cannot add connector '%s': A connector with the same name already exists",
+                connector));
+          });
+    }
+  }
+
+  private static Optional<String> checkConnectorExists(final ConnectClient client,
+      final CreateConnector createConnector) {
+    final ConnectResponse<List<String>> connectorsResponse = client.connectors();
+    if (connectorsResponse.datum().isPresent()) {
+      return connectorsResponse.datum()
+          .get()
+          .stream()
+          .filter(connector -> StringUtils.equalsIgnoreCase(createConnector.getName(), connector))
+          .findAny();
+    }
+    return Optional.empty();
+  }
```

Comment:
The `Optional<String>` return value of this is not needed. A simple `boolean` would suffice.  Also, it looks like you're ignoring any connect error.  If the response returns an error we probably want to bubble that up, right?   The code handles errors by doing this:

```java
return response.error()
        .map(err -> new ErrorEntity(statement.getStatementText(), err));
```

You'll need to do something similar if the `connectors` call fails.  You could do this by throwing an exception that holds the `ErrorEntity`, or by combining `checkConnectorExists` and `throwIfConnectorIsDuplicate` together and have `throwIfConnectorIsDuplicate` return an `Optional<ErrorEntity>`. Either works.

Reply:
Fixed this to throw the `ErrorEntity` if there occurs an error with `connects()`